lib/logstash/plugin_mixins/aws_config/v2.rb

```diff
@@ -27,42 +29,45 @@ def aws_options_hash
       opts[:region] = @region
     end
 
-    if !@endpoint.is_a?(NilClass)
-      opts[:endpoint] = @endpoint
-    end
+    opts[:endpoint] = @endpoint unless @endpoint.nil?
 
     return opts
   end
 
   private
-  def credentials
-    @creds ||= begin
-                 if @access_key_id && @secret_access_key
-                   credentials_opts = {
-                     :access_key_id => @access_key_id,
-                     :secret_access_key => @secret_access_key.value
-                   }
 
-                   credentials_opts[:session_token] = @session_token.value if @session_token
-                   Aws::Credentials.new(credentials_opts[:access_key_id],
-                                        credentials_opts[:secret_access_key],
-                                        credentials_opts[:session_token])
-                 elsif @aws_credentials_file
-                   credentials_opts = YAML.load_file(@aws_credentials_file)
-                   Aws::Credentials.new(credentials_opts[:access_key_id],
-                                        credentials_opts[:secret_access_key],
-                                        credentials_opts[:session_token])
-                 elsif @role_arn
-                   assume_role
-                 end
-               end
+  def aws_credentials
+    if @access_key_id && @secret_access_key
+      Aws::Credentials.new(@access_key_id, @secret_access_key.value, @session_token ? @session_token.value : nil)
+    elsif @access_key_id.nil? ^ @secret_access_key.nil?
+      @logger.warn("Likely config error: Only one of access_key_id or secret_access_key was provided but not both.")
+      secret_access_key = @secret_access_key ? @secret_access_key.value : nil
+      Aws::Credentials.new(@access_key_id, secret_access_key, @session_token ? @session_token.value : nil)
```

Comment:
not ideal but could work in some cases - previous behaviour was confusing as the set value was simply ignored